tests/unit_tests/test_server_client.py

```diff
@@ -737,6 +737,90 @@ def add_one(x: int) -> int:
         }
 
 
+@pytest.mark.asyncio
+async def test_astream_log_allowlist(event_loop: AbstractEventLoop) -> None:
+    """Test async stream with an allowlist."""
+
+    async def add_one(x: int) -> int:
+        """Add one to simulate a valid function"""
+        return x + 1
+
+    app = FastAPI()
+    add_routes(
+        app,
+        RunnableLambda(add_one).with_config({"run_name": "allowed"}),
+        path="/empty_allowlist",
+        input_type=int,
+        stream_log_name_allowlist=[],
+    )
+    add_routes(
+        app,
+        RunnableLambda(add_one).with_config({"run_name": "allowed"}),
+        input_type=int,
+        path="/allowlist",
+        stream_log_name_allowlist=["allowed"],
+    )
+
+    # Invoke request
+    async with get_async_remote_runnable(app, path="/empty_allowlist/") as runnable:
+        run_log_patches = []
+
+        async for chunk in runnable.astream_log(1):
+            run_log_patches.append(chunk)
+
+        assert len(run_log_patches) == 0
+
+        run_log_patches = []
+        async for chunk in runnable.astream_log(1, include_tags=[]):
+            run_log_patches.append(chunk)
+
+        assert len(run_log_patches) == 0
+
+        run_log_patches = []
+        async for chunk in runnable.astream_log(1, include_types=[]):
+            run_log_patches.append(chunk)
+
+        assert len(run_log_patches) == 0
+
+        run_log_patches = []
+        async for chunk in runnable.astream_log(1, include_names=[]):
+            run_log_patches.append(chunk)
+
+        assert len(run_log_patches) == 0
+
+    async with get_async_remote_runnable(app, path="/allowlist/") as runnable:
+        run_log_patches = []
+
+        async for chunk in runnable.astream_log(1):
+            run_log_patches.append(chunk)
+
+        assert len(run_log_patches) > 0
+
+        run_log_patches = []
+        async for chunk in runnable.astream_log(1, include_tags=[]):
+            run_log_patches.append(chunk)
+
+        assert len(run_log_patches) > 0
+
+        run_log_patches = []
+        async for chunk in runnable.astream_log(1, include_types=[]):
+            run_log_patches.append(chunk)
+
+        assert len(run_log_patches) > 0
```

Comment:
Not related to this PR 

---------------

I'm confused about semantics of `LogStreamCallbackHandler` --  `include_types=[]` is an empty list rather than a None... should we be getting no results in this case?